cpp/src/table/row_operators.cu

```diff
@@ -262,18 +330,25 @@ using column_checker_fn_t = std::function<void(column_view const&)>;
  */
 void check_lex_compatibility(table_view const& input)
 {
-  // Basically check if there's any LIST hiding anywhere in the table
+  // Basically check if there's any LIST of STRUCT or STRUCT of LIST hiding anywhere in the table
```

Comment:
Wait, so we only support these two structures? How about the rest, like `list<list<list<.....>>>`?

Reply:
We DON'T support these two structures. That's what we're checking for

Reply:
Why don't? I thought that now we will support arbitrarily nested types?

Reply:
That will be implemented in a separate PR. This PR only enables lexicographic comparison for pure lists. Perhaps you're confusing this with equality comparison which does support arbitrary nesting now.